my-vault/handlers/handlers.go

```diff
@@ -0,0 +1,164 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"github.com/google/uuid"
+	"sync"
+)
+
+type Secret struct {
+	Id    string 
+	Name  string  
+	KVMap map[string]string 
+	
+}
+var mu sync.Mutex
+var db = make(map[string]Secret)
+
+
+type CreateSecretRequest  struct {
+	Name  string    `json:"name"`
+	KVMap map[string]string `json:"kv_map"`
+	
+}
+
+type CreateSecretResponse  struct {
+	Message string `json:"message"`
+	Id string `json:"id"`
+	Name  string    `json:"name"`
+	
+	
+}
+type GetSecretResponse struct {
+	Message string `json:"message"`
+	Id    string  `json:"id"`
+	Name  string    `json:"name"`
+	KVMap map[string]string `json:"kv_map"`
+
+}
+
+func GenerateUUID() string{
+	newUUID, err := uuid.NewV7()
+	if err != nil {
+		fmt.Println("Error generating UUID:", err)
+		panic(err)
+	}
+	return  newUUID.String()
+}
+
+func NewSecretResponseObject(s Secret) GetSecretResponse {
+	return GetSecretResponse{
+		Message: "Retrieved secret object",
+		Id : s.Id,
+		Name: s.Name,
+		KVMap: s.KVMap,
+	}
+}
+
+// func NewCreateSecret(s CreateSecretRequest ) Secret {
+// 	return Secret{
+// 		Id: GenerateUUID(),
+// 		Name: s.Name,
+// 		KVMap: s.KVMap,
+
+// 	}
+// }
+
+func HandlePostSecret(w http.ResponseWriter, r *http.Request) {
+
+	
+	var secret Secret
+	var secretRequestObject CreateSecretRequest
+
+	err := json.NewDecoder(r.Body).Decode(&secretRequestObject)
+	
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
```

Comment:
I think it'll probably be useful to create a log message here for you as well so you can see when a bad request hits the server (if you intentionally didn't want to do this because it'll let an attacker spam you, that's fair).
```suggestion
		http.Error(w, err.Error(), http.StatusBadRequest)
		fmt.Println("Bad request:", err.Error())
		return
```